src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```diff
@@ -19,5 +19,30 @@ public static string TrimEnd(this string value, string suffix, StringComparison
                        ? value.Substring(0, value.Length - suffix.Length)
                        : value;
         }
+
+        /// <summary>
+        /// Converts a <see cref="string"/> into a <see cref="bool"/> by comparing it to commonly used values
+        /// such as "True", "yes", or "1". Case-insensitive. Defaults to <c>false</c> if string is not recognized.
+        /// </summary>
+        /// <param name="value">The string to convert.</param>
+        /// <returns><c>true</c> if <paramref name="value"/> is one of the accepted values for <c>true</c>; <c>false</c> otherwise.</returns>
+        public static bool? ToBoolean(this string value)
+        {
+            if (value == null) { throw new ArgumentNullException(nameof(value)); }
+
+            switch (value.ToUpperInvariant())
```

Comment:
This extension method is internal only. I'm using it to convert tags values like `span.SetSpan("force.keep", "true")` because `Span.SetSpan()` only allows string values. (Strongly-typed tags is something I would like to revisit later, but we need to support the OpenTracing convention regardless.) In the future, I plan to use the extension method to convert settings from string-typed sources like environment variables. In all cases, there will be only one correct documented value, but I wanted to give users a little margin for error by allowing more values aside from `true/false` when their intent is obvious. It is not meant to be exhaustive.

- I thought about allowing `Y/N`, `T/F`, and even `enable/disable`, but we have to draw the line somewhere. Once they are out there and people are using them, it will be harder to change or remove without breaking existing code.
- We shouldn't use `StartsWith()` because that would allow clearly invalid values like `Yesterday` or `Tomorrow`.
- Regarding `-1`, AFAIK the general rule is that `0` means `false` and any non-zero value means `true`. (not to be confused with return codes where `0` means success and non-zero values are error codes)
- The `switch` is effectively doing ordinal comparisons ignoring case, and in this case is much easier to read and maintain than a chain of `if/else` with `StringComparison`. I imagine that using `ToUpperInvariant()` once and making several case-sensitive comparisons is also slightly faster.